sound/soc/intel/boards/sof_ssp_amp.c

```diff
@@ -222,7 +232,7 @@ sof_card_dai_links_create(struct device *dev, enum sof_ssp_codec amp_type,
 			links[id].name = devm_kasprintf(dev, GFP_KERNEL, "SSP%d-HDMI", port);
 			if (!links[id].name)
 				return NULL;
-			links[id].id = id;
+			links[id].id = fixed_be ? (HDMI_IN_BE_ID + i - 1) : id;
```

Comment:
@brentlu can you explain when `fixed_be` is false here? It's defined `true` 15 lines above and never changed again.

Reply:
fixed_be will be false when the board config is 'adl_cs35l41'. Other board configs which implements HDMI-In are using the topology sof-tgl-rt1308-hdmi-ssp.m4 which is using fixed number for each BE link. I think there is still room for the code here to improve...

Reply:
Thanks for submitting follow-up https://github.com/thesofproject/linux/pull/4694 so quickly. Let's move all discussions there.